libcommon/touch.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2023 - Tillitis AB
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+
+#include <stdbool.h>
+#include <tkey/led.h>
+#include <tkey/touch.h>
+#include <tkey/qemu_debug.h>
+
+// CPU clock frequenzy in Hz
+#define CPUFREQ 18000000
+
+//#define timerrunning() (*timer_status & (1 << TK1_MMIO_TIMER_STATUS_RUNNING_BIT))
+
+// clang-format off
+static volatile uint32_t *timer =           (volatile uint32_t *)TK1_MMIO_TIMER_TIMER;
+static volatile uint32_t *timer_prescaler = (volatile uint32_t *)TK1_MMIO_TIMER_PRESCALER;
+static volatile uint32_t *timer_status =    (volatile uint32_t *)TK1_MMIO_TIMER_STATUS;
+static volatile uint32_t *timer_ctrl =      (volatile uint32_t *)TK1_MMIO_TIMER_CTRL;
+static volatile uint32_t *touch = (volatile uint32_t *)TK1_MMIO_TOUCH_STATUS;
+// clang-format on
+
+bool touchwait(int color, int timeout)
```

Comment:
We should have the unit on the name as well, like `timeout_ms` or `timeout_s` since we now actually support time as input.